docs/tutorial.py

```diff
@@ -0,0 +1,78 @@
+# ruff: noqa
+"""
+The kasa library is fully async and methods that perform IO need to be run inside an async couroutine.
+These examples assume async code is called from within `async def` or are running in a asyncio REPL
+(python -m asyncio)
+
+The main entry point for the api is :func:`~kasa.Discover.discover` and
+:func:`~kasa.Discover.discover_single` which return Device objects.
+Most newer devices require your tplink cloud username and password but this can be ommitted for older devices.
+
+>>> from kasa import Device, Discover
+
+:func:`~kasa.Discover.discover` returns a list of devices on your network
+
+>>> devices = await Discover.discover(username="user@mail.com", password="great_password")
+>>> for dev in devices:
+>>>     await dev.update()
+>>>     print(dev.host)
+127.0.0.1
+127.0.0.2
+
+:func:`~kasa.Discover.discover_single` returns a single device by hostname
+
+>>> dev = await Discover.discover_single("127.0.0.1", username="user@mail.com", password="great_password")
+>>> await dev.update()
+>>> dev.alias
+Living Room
+>>> dev.model
+L530
+>>> dev.rssi
+-52
+>>> dev.mac
+5C:E9:31:00:00:00
+>>> await dev.set_alias("Dining Room")
+>>> await dev.update()
+>>> dev.alias
+Dining Room
+
+Different groups of functionality are supported by modules which you can access with Module.Name.
```

Comment:
```suggestion
Different groups of functionality are supported by modules which you can access via `dev.modules`.
```
Maybe?

I'd split the first part about modules by start with showing all the available modules for the test device, and then continue how to access one of those using the names inside `Module`.

Reply:
I looked at that and it’s quite verbose so I was thinking maybe a link to a modules page and a feature page might be better?

Reply:
That would also work. Perhaps the "getting started" tutorial shows just the API basics, and modules & features can have their own examples (probably under the library design).

Reply:
Code suggestion committed. Will flesh out the module and features sections of the library design in a separate PR.